### ISSUE TITLE
More interesting mob AI

### ISSUE THREAD

**Issue by prophnoran** [prophecymud/ideas#12]
At 91, one the things we do is farm eq. This could be more interesting if the mobs fought back better: used heal potions, chased you around, ran away when injured, etc. There are some hard-coded versions of this in mobprogs.

Generally, making the PvE richer at 91 is a good idea, I think; at the moment there's no real reason to use teams, and no real danger of losing!

**Comment by anhegofcherek** [comment]
Agreed! This is one of those things I think will be significantly easier with GoProphecy and all it enables

Naradas and I did some initial work here: his mob memory code, so they'll return to their base position, rearm themselves after disarm etc.; my code so mobs will try to "go home" if moved; but as you point out, there's scope for so much more!

**Comment by anhegofcherek** [comment]
Do you have any specific thoughts/ideas here?

I've recently been playing around with pathing code, and I'm really tempted to do some kind of quest/feature with Chandim that hunt players around the game, but I'm struggling with actual concrete ideas

**Comment by prophnoran** [comment]
Sure, a couple things come to mind: 

the oyster trick shouldn't work (mobs do switch from fighting an oyster/ant to fighting you eventually, iirc; this should be more aggressive). 

If you flee, mobs should try to follow you. They should remember you if you come back into the same room and hit you back.

If a mob is badly injured, it should run away.

Mobs could have basic backstab AI.

Casters could have basic responses to what you do: if you're a backstabber, they should try to web. They can try to dispel you. They can switch what spell they cast according to vulns.

Broadly, imagine how the fight would go differently if even a moderately-competent player were controlling the mob instead; how much of that can we get? Even a few basics would help a lot, I think.

**Comment by anhegofcherek** [comment]
So, uh, turns out I completely broke mob AI when porting to Go, and just recently fixed that - before I make mobs any smarter, I'm going to give people a bit of time to recover.

For five months, mobs only landed one hit per round and didn't do anything smart like try to dispel, switch targets, dirt bsers, bash casters, etc. - now they're back on their game a bit more

These are good ideas, thank you - I'll let things settle a bit first!
